src/libcore/fmt/mod.rs

```diff
@@ -518,7 +518,8 @@ impl Display for Arguments<'_> {
     label="`{Self}` cannot be formatted using `{{:?}}` because it doesn't implement `{Debug}`",
 )]
 #[doc(alias = "{:?}")]
-#[lang = "debug_trait"]
+#[cfg_attr(bootstrap, lang = "debug_trait")]
+#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "debug_trait")]
```

Comment:
cc @petrochenkov I would probably prefer to call this `Debug`, but I guess other `rustc_diagnostic_item`s aren't uniform even?

Also, nowadays, should we change e.g. lang items to use an identifier instead of a string?

Reply:
I just kept with the naming scheme of lang items.

I'll address both for lang items in a parallel PR

Reply:
> nowadays, should we change e.g. lang items to use an identifier instead of a string?

That's not possible unless you want `lang(debug_trait)` instead of `lang = "debug_trait"`. After the `=` a literal is expected by rustc

Reply:
Oh, huh, I didn't realize it was limited to literals. Ignore me then.

Reply:
This'll probably need to be `boostrap_stdarch_ignore_this` instead of just `bootstrap`